carma-messenger-core/j2735_convertor/src/psm_convertor.cpp

```diff
@@ -0,0 +1,271 @@
+                                                             j2735_v2x_msgs::msg::AccelerationSet4Way::ACCELERATION_UNAVAILABLE);
+
+  // Convert Vertical
+  out_msg.vert =
+      ValueConvertor::valueCavToJ2735<int8_t>(in_msg.vert, units::FIFTIETH_G_PER_M_PER_SEC_SQR, in_msg.presence_vector,
+                                              carma_v2x_msgs::msg::AccelerationSet4Way::ACCELERATION_VERTICAL_AVAILABLE,
+                                              j2735_v2x_msgs::msg::AccelerationSet4Way::ACCELERATION_VERTICAL_UNAVAILABLE);
+
+  // Convert Yaw Rate
+  out_msg.yaw_rate = ValueConvertor::valueCavToJ2735<int16_t>(
+      in_msg.yaw_rate, units::CENTI_DEG_PER_DEG, in_msg.presence_vector,
+      carma_v2x_msgs::msg::AccelerationSet4Way::YAWRATE_AVAILABLE, j2735_v2x_msgs::msg::AccelerationSet4Way::YAWRATE_UNAVAILABLE);
+}
+
+void PSMConvertor::convert(const carma_v2x_msgs::msg::PositionalAccuracy& in_msg, j2735_v2x_msgs::msg::PositionalAccuracy& out_msg)
+{
+  // Convert semi_major Axis
+  out_msg.semi_major = ValueConvertor::valueCavToJ2735<uint8_t>(
+      in_msg.semi_major, units::TWENTIETH_M_PER_M, in_msg.presence_vector,
+      carma_v2x_msgs::msg::PositionalAccuracy::ACCURACY_AVAILABLE, j2735_v2x_msgs::msg::PositionalAccuracy::ACCURACY_UNAVAILABLE);
+
+  // Convert semi_minor Axis
+  out_msg.semi_minor = ValueConvertor::valueCavToJ2735<uint8_t>(
+      in_msg.semi_minor, units::TWENTIETH_M_PER_M, in_msg.presence_vector,
+      carma_v2x_msgs::msg::PositionalAccuracy::ACCURACY_AVAILABLE, j2735_v2x_msgs::msg::PositionalAccuracy::ACCURACY_UNAVAILABLE);
+
+  // Convert Orientation
+  out_msg.orientation = ValueConvertor::valueCavToJ2735<uint16_t>(
+      in_msg.orientation, units::DEG_360_OVER_65535_PER_DEG, in_msg.presence_vector,
+      carma_v2x_msgs::msg::PositionalAccuracy::ACCURACY_ORIENTATION_AVAILABLE,
+      j2735_v2x_msgs::msg::PositionalAccuracy::ACCURACY_ORIENTATION_UNAVAILABLE);
+}
+
+void PSMConvertor::convert(const carma_v2x_msgs::msg::PathHistory& in_msg, j2735_v2x_msgs::msg::PathHistory& out_msg)
+{
+    out_msg.curr_gnss_status = in_msg.curr_gnss_status;
+    out_msg.initial_position.lat.latitude = in_msg.initial_position.lat.latitude / latitude_conversion_const_;
+    out_msg.initial_position.lon.longitude = in_msg.initial_position.lon.longitude / longitude_conversion_const_;
+    out_msg.initial_position.elevation.elevation = in_msg.initial_position.elevation.elevation / elevation_conversion_const_;
+    out_msg.initial_position.utc_time = in_msg.initial_position.utc_time;
+    out_msg.initial_position.time_confidence = in_msg.initial_position.time_confidence;
+    out_msg.initial_position.pos_confidence = in_msg.initial_position.pos_confidence;
+    out_msg.initial_position.speed_confidence = in_msg.initial_position.speed_confidence;
+
+    convert(in_msg.initial_position.heading, out_msg.initial_position.heading);
+    convert(in_msg.initial_position.speed.speed, out_msg.initial_position.speed.speed);
```

Comment:
You are messing a transmission field here. That can be assigned directly 
out_msg.initial_position.speed.transmission = in_msg.initial_position.speed.transmission